nemo/collections/asr/parts/utils/nmesc_clustering.py

```diff
@@ -370,95 +365,97 @@ def getRepeatedList(mapping_argmat: torch.Tensor, score_mat_size: torch.Tensor):
     return repeat_list
 
 
-def get_argmin_mat(uniq_scale_dict: dict):
+@torch.jit.script
+def get_argmin_mat(timestamps_in_scales: List[torch.Tensor]):
     """
     Calculate the mapping between the base scale and other scales. A segment from a longer scale is
     repeatedly mapped to a segment from a shorter scale or the base scale.
 
     Args:
-        uniq_scale_dict (dict) :
-            Dictionary of embeddings and timestamps for each scale.
+        timestamps_in_scales (list):
+            List containing timestamp tensors for each scale.
+            Each tensor has dimensions of (Number of base segments) x 2.
 
     Returns:
-        session_scale_mapping_dict (dict) :
+        session_scale_mapping_dict (dict):
             Dictionary containing argmin arrays indexed by scale index.
     """
-    scale_list = sorted(list(uniq_scale_dict.keys()))
-    segment_anchor_dict = {}
+    scale_list = list(range(len(timestamps_in_scales)))
+    segment_anchor_list = []
     for scale_idx in scale_list:
-        time_stamp_list = uniq_scale_dict[scale_idx]['time_stamps']
-        time_stamps_float = torch.tensor([[float(x.split()[0]), float(x.split()[1])] for x in time_stamp_list])
-        segment_anchor_dict[scale_idx] = torch.mean(time_stamps_float, dim=1)
+        time_stamps_float = timestamps_in_scales[scale_idx]
+        segment_anchor_list.append(torch.mean(time_stamps_float, dim=1))
 
     base_scale_idx = max(scale_list)
-    base_scale_anchor = segment_anchor_dict[base_scale_idx]
-    session_scale_mapping_dict = {}
+    base_scale_anchor = segment_anchor_list[base_scale_idx]
+    session_scale_mapping_dict = []
     for scale_idx in scale_list:
-        curr_scale_anchor = segment_anchor_dict[scale_idx]
+        curr_scale_anchor = segment_anchor_list[scale_idx]
         curr_mat = torch.tile(curr_scale_anchor, (base_scale_anchor.shape[0], 1))
         base_mat = torch.tile(base_scale_anchor, (curr_scale_anchor.shape[0], 1)).t()
         argmin_mat = torch.argmin(torch.abs(curr_mat - base_mat), dim=1)
-        session_scale_mapping_dict[scale_idx] = argmin_mat
+        session_scale_mapping_dict.append(argmin_mat)
     return session_scale_mapping_dict
 
 
-def getMultiScaleCosAffinityMatrix(uniq_embs_and_timestamps: dict, device: torch.device = torch.device('cpu')):
+@torch.jit.script
+def getCosAffinityMatrix(emb: torch.Tensor):
```

Comment:
for jit script, is it not necessary to mention return data type?

Reply:
I tried return data type annotation but torch.jit.script compiler returned error on non-Tensor output. So I had to remove it.